### PR TITLE
refactor(trie): modify ReviveTrie to include shadow logic

### PR DESCRIPTION
**Pull request by asyukii**
### Description
add shadow logic to trie revive

### Changes

Notable changes: 
* modify ReviveTrie()
* fix MPT insert error

### CODE REVIEW

File: trie/trie.go [hunk 1]
```diff
@@ -865,29 +886,44 @@ func (t *Trie) ReviveTrie(proof []*MPTProofNub) (successNubs []*MPTProofNub) {
...
 	if len(successNubs) == 0 && len(proof) != 0 {
 		return successNubs, fmt.Errorf("all nubs failed to revive trie")
 	}
 
 	return successNubs, nil
 }
 
-func (t *Trie) tryRevive(n node, key []byte, nub MPTProofNub) (node, bool, error) {
-	if len(key) == 0 {
+func (t *Trie) tryRevive(n node, key []byte, nub MPTProofNub, path []byte, isExpired bool) (node, bool, error) {
```
Comment (0xbundler) [inline trie/trie.go:916]: Where is the parameter 'path' used?

File: trie/trie.go [hunk 2]
```diff
@@ -899,34 +935,74 @@ func (t *Trie) tryRevive(n node, key []byte, nub MPTProofNub) (node, bool, error
...
+
 		return n, didResolve, err
 	case hashNode:
-		tn, err := t.resolveHash(n, nil) // TODO (asyukii): Revisit epoch index
+		tn, err := t.resolveHash(n, nil)
 		if err != nil {
 			return nil, false, err
 		}
-		return t.tryRevive(tn, key, nub)
+		return t.tryRevive(tn, key, nub, path, isExpired)
```
Comment (0xbundler) [inline trie/trie.go:1005]: May miss didResolve? In revive, didResolve mean resolveHash or reviveTrieNode?

File: trie/trie.go (hunk 1 above)
Comment (asyukii) [reply trie/trie.go:916]: 'path' is used only in ExpiredNodeError, returning the partial path to the expired node

File: trie/trie.go (hunk 2 above)
Comment (asyukii) [reply trie/trie.go:1005]: in this context, didResolve means reviveTrieNode.

changed "didResolve" to "didRevive" for better understanding

Comment (0xbundler) [reply trie/trie.go:971]: but when it resolves HashNode, there needs copy too?

Comment (asyukii) [reply trie/trie.go]: it will only copy the resolved hash node if the nub is revived successfully

**Review by 0xbundler — APPROVED** [review]

**Pull request merged**